src/adapters/mqtt/index.ts

```diff
@@ -168,6 +170,17 @@ class MqttAdapter extends Adapter {
     })
   }
 
+  private async resolveAuthConfig() {
```

Comment:
The `resolveAuthConfig()` function logic is repeated in all of the adapters. I feel like we can make it a part of the `Adapter` class.

Reply:
Different protocols have different config types, I tried writing it in adapters logic but it was a lot of nested if statements. I think it will be confusing when support for new protocols are added. But if we implement it on the protocol-specific adapters it would be easier as you don't need to know about other protocol config types.

Reply:
I see. then maybe it's time to make a few changes to the `configs.ts` file itself. I think it should be a single truth source for all config-related things. we can abstract all of the logic there with a method like `getServerConfig(serverName: string)`. 🤔   WDYT?

Reply:
are you suggesting that we move the config to the server name instead of protocol name so the config file would look something like this 

```ts
export default async function () {
  return {
    'server-name': {
      protocol: 'ws'
      // rest of the config. 
    }
  }
}

```

should we move this to an issue and explore the possibilities.

Reply:
I think the current format of the config file makes sense. All I am saying is that we should abstract the config parsing logic for different protocols and servers to https://github.com/asyncapi/glee/blob/master/src/lib/configs.ts

Reply:
Now I got it, sure I will try this and let you know.

Reply:
@KhudaDad414 what about now, I have moved the parsing logic to the adapter class.